smac/utils/sklearn_wrapper.py

```diff
@@ -0,0 +1,392 @@
+from sklearn.metrics.scorer import _check_multimetric_scoring
+from sklearn.model_selection._split import check_cv, BaseCrossValidator
+from sklearn.model_selection._search import BaseSearchCV
+from sklearn.model_selection._validation import _fit_and_score
+from sklearn.utils.fixes import MaskedArray
+from sklearn.utils.validation import indexable
+
+from smac.configspace import ConfigurationSpace
+from smac.facade.smac_facade import SMAC
+from smac.scenario.scenario import Scenario
+
+
+class ModelBasedOptimization(BaseSearchCV):
+    """
+    Scikit-Learn wrapper for SMAC
+    """
+
+    def __init__(self, estimator: BaseEstimator, param_distributions: Dict[str, List], n_iter: int = 200,
+                 # scoring: Union[str, Callable, None] = None, NOT SUPPORTED FOR NOW
+                 fit_params: Union[Dict, None] = None, iid: bool = True,
+                 refit: bool = True, cv: Union[BaseCrossValidator, int] = None, verbose: int = 0,
+                 random_state: Union[int, np.random.RandomState] = None, error_score: Union[str, int] = 'raise',
+                 return_train_score: bool = True, intensification_percentage: Union[float, None] = None):
+        """Scikit-learn wrapper for Sequential Model Based Optimization (SMAC). This allows to use SMAC together
+        with various tools relying on the scikit-learn API, such as OpenML.org.
+
+        Works similar to the RandomizedSearchCV class. For detailed and up-to-date
+        parameter descriptions, please see:
+        http://scikit-learn.org/stable/modules/generated/sklearn.model_selection.RandomizedSearchCV.html
+
+        Main difference: requires dict of lists for param_distributions (whereas the base-class
+        is more liberal and also allows for scipy distributions).
+
+        Parameters
+        ---------
+        estimator : estimator object.
+            A scikit-learn classifier. Should inherit from
+            sklearn.base.BaseEstimator. Either estimator needs
+            to provide a ``score`` function, or ``scoring`` must be passed.
+
+        param_distributions : dict
+            Dictionary with parameters names (string) as keys and
+            lists of parameters to try. Distributions must provide a ``rvs``
+            method for sampling (such as those from scipy.stats.distributions).
+            If a list is given, it is sampled uniformly.
+
+        n_iter : int, default=10
+            Number of parameter settings that are sampled. n_iter trades
+            off runtime vs quality of the solution.
+
+        scoring : string, callable or None, default=None
+            A string (see model evaluation documentation) or
+            a scorer callable object / function with signature
+            ``scorer(estimator, X, y)``.
+            If ``None``, the ``score`` method of the estimator is used.
+
+        fit_params : dict, optional
+            Deprecated feature of Scikit-learn
+
+        iid : boolean, default=True
```

Comment:
This one will actually be deprecated ;) But soon we'll also have https://github.com/scikit-learn/scikit-learn/pull/9599 which will make everything easier!